src/app.py

```diff
@@ -1,73 +1,354 @@
+
+    TODO:
+        - sanity check timestamp?
+        - convert datetime to datetime
+    """
+    data = request.json
+    failed, success = [], []
+    if data:
+        # this will only give the data that "exists" in the table
+        query = CatalogueItem.query.filter(CatalogueItem.uuid.in_(data.keys()))
+        for d in query:
+            try:
+                d.update(data.get(d.uuid))
+                success.append(d.uuid)
+            except:
+                failed.append(d.uuid)
+    db.session.commit()
+
+    # get all thoese keys that weren't updated in the db
+    failed = list(set(data.keys()) - set(success))
+    logger.debug(f"success: {len(success)} | failed: {len(failed)}")
+
+    return jsonify(dict(failed=failed, succes=success))
+
+
+@app.route("/catalogue/bulk/csv/", methods=["POST"])
+@token_required
 def upload_csv():
+    """
+    Endpoint to upload CSV and update catalogeitem table
+
+    This only accepts a csv file with following columns (strictly):
+        - Id (unique identifier to the file)
+        - Name
+        - ContentLength
+        - IngestionDate
+        - ContentDate:Start
+        - ContentDate:End
+        - Checksum:Algorithm
+        - Checksum:Value
+
+    TODO:
+        - optimize csv loader for a very large CSV
+        - optimize csv dump to database
+        - async upload
+    """
     logger.info("/catalogue/upload/ POST called")
     fname = uuid.uuid4().hex
     fpath = os.path.join("tmp", f"{fname}.csv")
     body = request.files["csv"]
     body.save(fpath)
 
     logger.debug("Normalizing column names...")
-    data = pd.read_csv(fpath)
-    data.rename(
-        columns={
-            "Id": "uuid",
-            "Name": "name",
-            "ContentLength": "content_length",
-            "IngestionDate": "ingestion_date",
-            "ContentDate:Start": "content_date_start",
-            "ContentDate:End": "content_date_end",
-            "Checksum:Algorithm": "checksum_algorithm",
-            "Checksum:Value": "checksum_value",
-        },
-        inplace=True,
-    )
-    logger.debug("Dumping to table={CatalogueItem.__tablename__}")
-    _ = data.to_sql(
-        name=CatalogueItem.__tablename__,
-        con=db.engine,
-        index=False,
-        if_exists="replace",
-    )
-    logger.debug("Successful!")
-    return "success"
+    try:
+        data = pd.read_csv(fpath)
+    except:
+        logger.error("Failed to load csv")
+        clean_files([fpath])
+        abort_json(400, error="INVALID_FILE")
+
+    try:
+        data.rename(
+            columns={
+                "Id": "uuid",
+                "Name": "name",
+                "ContentLength": "content_length",
+                "IngestionDate": "ingestion_date",
+                "ContentDate:Start": "content_date_start",
+                "ContentDate:End": "content_date_end",
+                "Checksum:Algorithm": "checksum_algorithm",
+                "Checksum:Value": "checksum_value",
+            },
```

Comment:
Move to a constant.

Reply:
Do you think we should put it as a part of config or separately as a part of global constants (say in `constants.py`).

Reply:
Just in this file is fine, as this is only needed here.